app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -189,185 +190,263 @@ public function getCartDataJs($checkoutType = 'multi-page')
-            if (Mage::registry("api_error")) {
-                $cartData['error'] = Mage::registry("api_error");
-            }
+        } catch (Exception $e) {
+            Mage::helper('boltpay/bugsnag')->notifyException($e);
+        }
+    }
 
-            $jsonCart = json_encode($cartData);
-            $jsonHints = '{}';
-            if (sizeof($hintData) != 0) {
-                // Convert $hint_data to object, because when empty data it consists array not an object
-                $jsonHints = json_encode($hintData, JSON_FORCE_OBJECT);
-            }
+    /**
+     * @param $checkoutType
+     * @return bool
+     */
+    protected function isAdminAndUseJsInAdmin($checkoutType)
+    {
+        return ($checkoutType === self::CHECKOUT_TYPE_ADMIN) && !Mage::getStoreConfig('payment/boltpay/use_javascript_in_admin');
+    }
 
-            //////////////////////////////////////////////////////////////////////////
-            // Format the success and save order urls for the javascript returned below.
-            $successUrl   = $this->getUrl(Mage::getStoreConfig('payment/boltpay/successpage'));
-            $saveOrderUrl = $this->getUrl('boltpay/order/save');
-
-            //////////////////////////////////////////////////////
-            // Collect the event Javascripts
-            // We execute these events as early as possible, typically
-            // before Bolt defined event JS to give merchants the
-            // opportunity to do full overrides
-            //////////////////////////////////////////////////////
-            if (($checkoutType === 'admin') && !Mage::getStoreConfig('payment/boltpay/use_javascript_in_admin')) {
-                $check = '';
-                $onCheckoutStart = '';
-                $onShippingDetailsComplete = '';
-                $onShippingOptionsComplete = '';
-                $onPaymentSubmit = '';
-                $success = '';
-                $close = '';
-            } else {
-                $check = Mage::getStoreConfig('payment/boltpay/check');
-                $onCheckoutStart = Mage::getStoreConfig('payment/boltpay/on_checkout_start');
-                $onShippingDetailsComplete = Mage::getStoreConfig('payment/boltpay/on_shipping_details_complete');
-                $onShippingOptionsComplete = Mage::getStoreConfig('payment/boltpay/on_shipping_options_complete');
-                $onPaymentSubmit = Mage::getStoreConfig('payment/boltpay/on_payment_submit');
-                $success = Mage::getStoreConfig('payment/boltpay/success');
-                $close = Mage::getStoreConfig('payment/boltpay/close');
-            }
+    /**
+     * Generate cart data
+     *
+     * @param $orderCreationResponse
+     * @return array
+     */
+    public function buildCartData($orderCreationResponse)
+    {
+        $authCapture = (Mage::getStoreConfigFlag('payment/boltpay/auto_capture') === self::AUTO_CAPTURE_ENABLED);
```

Comment:
why not simply `Mage::getStoreConfigFlag('payment/boltpay/auto_capture')` ?

Reply:
agree

Reply:
Previously it was 1 or 0. and it cause the different types. I will refactor it now.